### PR TITLE
Added third type of JS mixin

### PR DESCRIPTION
**Pull request by drpayyne**
## Purpose of this pull request
This pull request (PR) adds the third type of Javascript mixins for modules which return objects, instead of Ui Components or jQuery widgets. Example module: [Magento_Checkout/.../step-navigator.js](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Checkout/view/frontend/web/js/model/step-navigator.js)
- Added third type of JS mixin for objects
- Added references and links

## Affected DevDocs pages
- https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/javascript/js_mixins.html
- https://devdocs.magento.com/guides/v2.2/javascript-dev-guide/javascript/js_mixins.html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/javascript-dev-guide/javascript/js_mixins.md [hunk 1]
```diff
@@ -133,6 +156,22 @@ var config = {
 };
 ```
 
+The following is an example of a `requirejs-config.js` file that adds the `step-navigator-mixin` mixin defined in the previous example, to the [step navigator object][].
+
+**File:** `OrangeCompany/Sample/view/frontend/requirejs-config.js`
+
+```javascript
+var config = {
```
Comment (rogyar) [inline guides/v2.2/javascript-dev-guide/javascript/js_mixins.md]: Please, merge this example with the previous one. That's fair that it's applicable only for the frontend area. However, multiplying similar examples might be an overhead for the reader. So, in this particular case, we can move this declaration to the base area (just add the declaration to the previous example)

File: guides/v2.2/javascript-dev-guide/javascript/js_mixins.md [hunk 2]
```diff
@@ -105,6 +105,29 @@ define(['jquery'], function ($) {
 });
 ```
 
+#### Extend JS Object
+
+The last kind of JS mixin is when the base Javscript file returns an object. In this case, a wrapper is necessary. The following is an example of a mixin that extends the `setHash` function of [step navigator object][]. Here, `_super()` is the base method which can be called if needed.
```
Comment (rogyar) [inline guides/v2.2/javascript-dev-guide/javascript/js_mixins.md]: To make this statement more future-proof, let's adjust the description slightly:

```suggestion
Another use-case of JS mixin is when the base Javascript file returns an object. In this case, a wrapper is necessary. The following is an example of a mixin that extends the `setHash` function of [step navigator object][]. Here, `_super()` is the base method that can be called if needed.
```

Thank you!

File: guides/v2.2/javascript-dev-guide/javascript/js_mixins.md (hunk 1 above)
Comment (drpayyne) [reply guides/v2.2/javascript-dev-guide/javascript/js_mixins.md]: Fixed it, thank you!

File: guides/v2.2/javascript-dev-guide/javascript/js_mixins.md [hunk 3]
```diff
@@ -105,6 +105,29 @@ define(['jquery'], function ($) {
 });
 ```
 
+#### Extend JS Object
+
+Another use-case of JS mixin is when the base Javascript file returns an object. In this case, a wrapper is necessary. The following is an example of a mixin that extends the `setHash` function of [step navigator object][]. Here, `_super()` is the base method that can be called if needed.
```
Comment (keharper) [inline guides/v2.2/javascript-dev-guide/javascript/js_mixins.md]:
```suggestion
Another use-case for the JS mixin is when the base Javascript file returns an object. In this case, a wrapper is necessary. The following example mixin extends the `setHash` function of [step navigator object][]. Here, `_super()` is the base method that can be called if needed.
```

File: guides/v2.2/javascript-dev-guide/javascript/js_mixins.md [hunk 4]
```diff
@@ -114,7 +137,7 @@ The mixins configuration in the `requirejs-config.js` associates a target compon
 
 ### Example
 
-The following is an example of a `requirejs-config.js` file that adds the `columns-mixin` and `modal-widget-mixin` mixins, defined in the previous examples, to the [grid column component][] and [modal widget][].
+The following is an example of a `requirejs-config.js` file that adds the `columns-mixin`, `modal-widget-mixin`, and `step-navigator-mixin` mixins, defined in the previous examples, to the [grid column component][], [modal widget][], and [step navigator object][].
```
Comment (keharper) [inline guides/v2.2/javascript-dev-guide/javascript/js_mixins.md]:
```suggestion
The following is an example of a `requirejs-config.js` file that adds the `columns-mixin`, `modal-widget-mixin`, and `step-navigator-mixin` mixins, which were defined in the previous examples, to the [grid column component][], [modal widget][], and [step navigator object][].
```

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
@drpayyne This PR is failing the linting tests. Please remove the blank space at the end of line 120 and all the spaces on line 125.

**Comment by drpayyne** [comment]
Fixed it, @keharper. Thank you

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
Thanks @drpayyne

**Pull request merged**